### PR TITLE
Add ability to set and unset docs preferences

### PR DESCRIPTION
**Pull request by joelzwarrington**
### Summary and motivation

Adds `stripe docs prefs set` and `stripe docs prefs unset` commands to persist documentation preferences locally. Also updates `stripe docs prefs list` to show the currently configured value for each preference.

Preferences are stored in the CLI config file under the active profile, so they carry over between sessions. `set` validates the ID and value against the live list from docs.stripe.com before writing.

### Test plan

- [X] Changes are covered by unit tests (including failures and edge cases)
- [X] Changes were tested manually

To test manually:

```bash
make
./stripe docs prefs list
./stripe docs prefs set server go
./stripe docs prefs list   # should show [current: go] next to server
./stripe docs prefs unset server
./stripe docs prefs list   # [current: ...] should be gone
```

You should see a styled success message on set/unset (`✓ Preference server set to go`), and the list should reflect the current value inline with each preference's allowed values.

CleanShot 2026-07-21 at 10 48 30

### CODE REVIEW

File: pkg/cmd/docs/prefs_test.go [hunk 1]
```diff
@@ -112,3 +137,109 @@ func TestPrefsListCommand_ServerError(t *testing.T) {
...
+	client := docs.NewClient("test").WithOptions(docs.WithBaseURL("http://unused"))
+	var out bytes.Buffer
+	root := cmd.New().WithOptions(cmd.WithClient(client), cmd.WithConfig(cfg)).Root()
+	root.SetOut(&out)
+	root.SetArgs([]string{"prefs", "unset", "lang"})
+
+	require.NoError(t, root.ExecuteContext(context.Background()))
+	assert.Contains(t, stripANSI(out.String()), "Preference lang unset")
+	assert.Equal(t, "", viper.GetString("default.docs_prefs.lang"))
+}
```
Comment (rayc-stripe) [inline pkg/cmd/docs/prefs_test.go:245]: Can we add some failure cases for the unset command? What should we see if a pref is invalid or the value is not set but unset is run?

**Review by rayc-stripe — COMMENTED** [review]
out-of-scope question: do we have a plan to set the pref for logged-in users by default (if they have some prefs stored on the server)?

File: pkg/cmd/docs/prefs_test.go (hunk 1 above)
Comment (joelzwarrington) [reply pkg/cmd/docs/prefs_test.go:245]: I added more tests which show that a failure isn't really possible and they're just no-ops.

Comment (rayc-stripe) [reply pkg/cmd/docs/prefs_test.go:245]: should we consider showing some error/warning message when the unset command hits a invalid value?

**Review by rayc-stripe — APPROVED** [review]
